### PR TITLE
Add ability to enable cache mutation detector in GCE

### PR DESCRIPTION
**Pull request by ncdc**
Add the ability to enable the cache mutation detector in GCE. The current default behavior (disabled) is retained.

When paired with https://github.com/kubernetes/test-infra/pull/1901, we'll be able to detect shared informer cache mutations in gce e2e PR jobs.

### CODE REVIEW

**Comment by ncdc** [comment]
cc @deads2k @gmarek @wojtek-t @smarterclayton 

@krzyzacy this is the PR for what we discussed in Slack the other day. PTAL and let me know if I need to correct anything. Thanks!

**Comment by ncdc** [comment]
@cjcullen can you help determine if the GKE failure was related to this PR?

**Comment by smarterclayton** [comment]
Note that this causes a significant performance hit so we should double check it isn't accidentally enabled on any Kubemark test

**Comment by ncdc** [comment]
> Note that this causes a significant performance hit so we should double check it isn't accidentally enabled on any Kubemark test

@smarterclayton that's why I didn't apply this to the Kubemark scripts in https://github.com/kubernetes/test-infra/pull/1901

File: cluster/gce/container-linux/configure-helper.sh [hunk 1]
```diff
@@ -567,6 +567,10 @@ function start-kubelet {
   local -r kubelet_env_file="/etc/kubelet-env"
   echo "KUBELET_OPTS=\"${flags}\"" > "${kubelet_env_file}"
 
+  if [[ -n "${ENABLE_CACHE_MUTATION_DETECTOR:-}" ]]; then
+		echo "KUBE_CACHE_MUTATION_DETECTOR=true" >> "${kubelet_env_file}"
```
Comment (mikedanese) [inline cluster/gce/container-linux/configure-helper.sh]: please use double spaces for indentation.

Comment (ncdc) [inline cluster/gce/container-linux/configure-helper.sh]: Whoa, no idea how that happened. Will fix.

**Comment by ncdc** [comment]
Clearly my changes have busted things. I'm trying to figure out why, but if anyone has more familiarity with how the gce cluster is bootstrapped and could possibly quickly identify the problem, I'd appreciate the help.

File: cluster/gce/gci/configure-helper.sh [hunk 2]
```diff
@@ -990,6 +995,7 @@ function start-kube-apiserver {
   sed -i -e "s@{{admission_controller_config_volume}}@${admission_controller_config_volume}@g" "${src_file}"
   sed -i -e "s@{{image_policy_webhook_config_mount}}@${image_policy_webhook_config_mount}@g" "${src_file}"
   sed -i -e "s@{{image_policy_webhook_config_volume}}@${image_policy_webhook_config_volume}@g" "${src_file}"
+  sed -i -e "s@{{pillar\['enable_cache_mutation_detector'\]}}@${ENABLE_CACHE_MUTATION_DETECTOR}@g" "${src_file}"
```
Comment (cjcullen) [inline cluster/gce/gci/configure-helper.sh]: This script will die right here if the variable is undefined.

Comment (ncdc) [inline cluster/gce/gci/configure-helper.sh]: @cjcullen that sounds like it might be it. Where is the right place to define it?

Comment (ncdc) [inline cluster/gce/gci/configure-helper.sh]: Although the apiserver started in at least 1 of the jobs

Comment (ncdc) [inline cluster/gce/gci/configure-helper.sh]: One of the failing jobs couldn't get all of the system pods running (e.g. `event for fluentd-gcp-v1.38-tf9jx: {kubelet e2e-gce-agent-pr-43-0-minion-group-wm8d} FailedSync: Error syncing pod, skipping: network is not ready: [runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:docker: network plugin is not ready: Kubenet does not have netConfig. This is most likely due to lack of PodCIDR]`)

Comment (cjcullen) [inline cluster/gce/gci/configure-helper.sh]: I think you can follow the pattern of the `kube_docker_registry` temporary variable above.

**Comment by ncdc** [comment]
@k8s-bot gce etcd3 e2e test this, might be #37704 / #39108

**Comment by ncdc** [comment]
@k8s-bot gci gce e2e test this, some of the pods are having trouble starting for some reason (e.g. docker's network isn't set up). Let's see if it's just a flake.

**Comment by ncdc** [comment]
@k8s-bot kubemark e2e test this, more network plugin not ready issues.

**Comment by ncdc** [comment]
@k8s-bot kubemark e2e test this

**Comment by ncdc** [comment]
@k8s-bot gci gce e2e test this

**Comment by ncdc** [comment]
@k8s-bot kubemark e2e test this

**Comment by ncdc** [comment]
Finally green! @mikedanese would you mind taking another look? Thanks!

**Pull request merged**